app/services/pipeline_report_service.rb

```diff
@@ -67,13 +67,39 @@ def call
   end
 
   def generate
-    pipeline_run = PipelineRun.find(@pipeline_run_id)
     if @pipeline_run_id.nil?
-      Rails.logger.error("Pipeline run doesn't exist.")
-      return ""
-    elsif pipeline_run.total_reads.nil? || pipeline_run.adjusted_remaining_reads.nil?
-      Rails.logger.error("Pipeline run #{@pipeline_run_id} has no reads.")
-      return ""
+      return JSON.dump(
```

Comment:
Let's add this to a function called `get_pipeline_status` which can get the pipeline run.

Reply:
Done.